.gemini/styleguide.md

```diff
@@ -0,0 +1,749 @@
+    the data used for training. This task can be used for pre-training or
+    fine-tuning a MyModel, simply by calling `fit()`.
+
+    This model has a `generate()` method, which generates text based on a
+    prompt. The generation strategy used is controlled by an additional
+    `sampler` argument on `compile()`. You can recompile the model with
+    different `keras_hub.samplers` objects to control the generation. By
+    default, `"greedy"` sampling will be used.
+
+    This model can optionally be configured with a `preprocessor` layer, in
+    which case it will automatically apply preprocessing to string inputs during
+    `fit()`, `predict()`, `evaluate()` and `generate()`. This is done by default
+    when creating the model with `from_preset()`.
+
+    Args:
+        backbone: A `keras_hub.models.MyModelBackbone` instance.
+        preprocessor: A `keras_hub.models.MyModelCausalLMPreprocessor` or `None`.
+            If `None`, this model will not apply preprocessing, and inputs
+            should be preprocessed before calling the model.
+
+    Examples:
+
+    Use `generate()` to do text generation.
+    ```python
+    model = keras_hub.models.MyModelCausalLM.from_preset("my_model_base")
+    model.generate("I want to say", max_length=30)
+
+    # Generate with batched prompts.
+    model.generate(["This is a", "Where are you"], max_length=30)
+    ```
+
+    Compile the `generate()` function with a custom sampler.
+    ```python
+    model = keras_hub.models.MyModelCausalLM.from_preset("my_model_base")
+    model.compile(sampler="top_k")
+    model.generate("I want to say", max_length=30)
+
+    model.compile(sampler=keras_hub.samplers.BeamSampler(num_beams=2))
+    model.generate("I want to say", max_length=30)
+    ```
+
+    Call `fit()` on a single batch.
+    ```python
+    features = ["The quick brown fox jumped.", "I forgot my homework."]
+    model = keras_hub.models.MyModelCausalLM.from_preset("my_model_base")
+    model.fit(x=features, batch_size=2)
+    ```
+    """
+
+    backbone_cls = MyModelBackbone
+    preprocessor_cls = MyModelPreprocessor
+
+    def __init__(
+        self,
+        backbone,
+        preprocessor=None,
+        **kwargs,
+    ):
+        # === Layers ===
+        self.backbone = backbone
+        self.preprocessor = preprocessor
+
+        # === Functional Model ===
+        inputs = backbone.input
+        hidden_states = backbone(inputs)
+        outputs = backbone.token_embedding(hidden_states, reverse=True)
+        super().__init__(
+            inputs=inputs,
+            outputs=outputs,
+            **kwargs,
+        )
+```
+
+### Presets (`<model_name>_presets.py`)
+
+**Purpose**: This file defines a dictionary of preset configurations for the model.
+
+**Content**: Each entry includes the configuration arguments for the model (`config`), a description, and the URL to the pre-trained weights hosted on Kaggle (`weights_url`).
```

Comment:
Can we add extra line to make sure the preset names are Keras standard format, separated by underscores and only accepted special characters instead of "-" etc.